commit 69e71ace804a4055eff64b581e6b7bb465abf1a8
Author: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date:   Thu Oct 2 15:53:24 2014 +0200

    Fix linking issue on some platforms when building against gstreamer 1.0
    
    We need ifdefs if PlatformVideoWindowPrivate.h so that moc does not
    generate
    code references methods not build because they are ifdef'ed out in
    PlatformVideoWindowQt.cpp.

diff --git a/Source/WebCore/platform/graphics/gstreamer/PlatformVideoWindowPrivate.h b/Source/WebCore/platform/graphics/gstreamer/PlatformVideoWindowPrivate.h
index 8e28feb..e243771 100644
--- a/Source/WebCore/platform/graphics/gstreamer/PlatformVideoWindowPrivate.h
+++ b/Source/WebCore/platform/graphics/gstreamer/PlatformVideoWindowPrivate.h
@@ -19,6 +19,8 @@
 
 #ifndef PlatformVideoWindowPrivate_h
 #define PlatformVideoWindowPrivate_h
+// We do not use ENABLE or USE because moc does not expand these macros.
+#if defined(WTF_USE_GSTREAMER) && WTF_USE_GSTREAMER && !defined(GST_API_VERSION_1)
 
 #include <QEvent>
 #include <QTimer>
@@ -65,5 +67,6 @@ private:
 
 } // namespace WebCore
 
+#endif // USE(GSTREAMER) && !defined(GST_API_VERSION_1)
 
 #endif // PlatformVideoWindowPrivate_h
diff --git a/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.cpp b/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.cpp
index 3b244c6..c488e02 100644
--- a/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/FullScreenVideoQt.cpp
@@ -30,7 +30,7 @@
 #include "HTMLVideoElement.h"
 #include "Node.h"
 
-#if USE(GSTREAMER)
+#if USE(GSTREAMER) && !defined(GST_API_VERSION_1)
 #include "GStreamerGWorld.h"
 #include "PlatformVideoWindowPrivate.h"
 #endif
